File: 2019/024-nim/fib.nim
#? replace(sub = "\t", by = "  ")

from times import cpuTime

proc clock() : float = cpuTime()


# proc fib(n: uint64): uint64 = 
# 	if n < 2.uint64: 1.uint64 else: fib(n - 1) + fib(n - 2)

proc fib(n: int): uint64 = 
	var a,b : uint64 
	(a,b) = (1.uint64, 1.uint64)

	for i in 1..<n:
		(a,b) = (b,a+b)
	b

let start = clock()
var z : uint64
for i in 0..<100000000:
	z = fib 46
echo z
echo clock()-start

File: 2019/024-nim/fib.cpp
#include <iostream>
#include <ctime>

using namespace std;

// uint64_t fib(int n) {
// 	if (n < 2) {
// 		return 1;
// 	} else {
// 		return fib(n - 1) + fib(n - 2);
// 	}
// }  

uint64_t fib(int n) {
	uint64_t a=1,b=1;

	for (int i=0; i<n; i++) {
		uint64_t tmp = a;
		a = b;
		b = tmp+b;
	}
	return a;
}


int main() {
	int start = clock();
	uint64_t z = 0;
	for (int i=0; i<1000000000; i++) {
		z=fib(46);
	}
	cout << z << ' ';
	cout << clock()-start;
}
